config/seed-peer-generator/src/main.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use libra_config::config::{PersistableConfig, HANDSHAKE_VERSION};
+use libra_network_address::NetworkAddress;
+use libra_secure_json_rpc::JsonRpcClient;
+use libra_types::{on_chain_config::ValidatorSet, validator_info::ValidatorInfo, PeerId};
+use std::{collections::HashMap, convert::TryFrom, path::PathBuf};
+use structopt::StructOpt;
+use libra_types::account_address::AccountAddress;
+
+// TODO: Use the definition from network?
+pub type SeedPeersConfig = HashMap<PeerId, Vec<NetworkAddress>>;
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Debug, StructOpt)]
+#[structopt(about = "Tool to generate configs from chain")]
+enum Args {
+    #[structopt(about = "Generate seed peers config")]
+    GenerateSeedPeers(GenerateArgs),
+}
+
+#[derive(Debug, StructOpt)]
+struct GenerateArgs {
+    #[structopt(short = "o", long, parse(from_os_str))]
+    /// The output directory
+    output_dir: PathBuf,
+    #[structopt(short = "e", long)]
+    /// JSON RPC endpoint
+    endpoint: String,
+    #[structopt(short = "a", long)]
+    /// Specify the AccountAddress to retrieve peers for
+    account_address: AccountAddress,
+}
+
+fn main() {
+    let args = Args::from_args();
+
+    match args {
+        Args::GenerateSeedPeers(args) => generate_seed_peers_config(args)
+    };
+}
+
+fn generate_seed_peers_config(args: GenerateArgs) {
+    let mut seed_peers_config = SeedPeersConfig::new();
+    let validator_set = get_validator_set(args.endpoint, args.peer_id).unwrap();
+
+    // Add the validators to the config!
+    if let Some(validator_set) = validator_set {
+        for validator in validator_set.payload().iter() {
+            let (peer_id, seed_addrs) = to_seed_peer(validator);
+            seed_peers_config.insert(peer_id, seed_addrs);
+        }
```

Comment:
Maybe we should get rid of this SeedPeersConfig and use a type statement instead.. ahh but not here :)

Reply:
I'm not sure what you mean.  SeedPeersConfig is just a type named of a specific map.